dtfs-central-api/src/services/repositories/utilisation-reports-repo/utilisation-reports-repo.ts

```diff
@@ -26,9 +26,16 @@ export const saveUtilisationReportDetails = async (reportPeriod: ReportPeriod, a
     },
   };
 
+  const filterForReportInReportPeriod = {
+    'reportPeriod.start.month': reportPeriod.start.month,
+    'reportPeriod.start.year': reportPeriod.start.year,
+    'reportPeriod.end.month': reportPeriod.end.month,
+    'reportPeriod.end.year': reportPeriod.end.year,
+  };
+
   const utilisationReportDetailsCollection = await db.getCollection(DB_COLLECTIONS.UTILISATION_REPORTS);
-  const savedDetails = await utilisationReportDetailsCollection.insertOne(utilisationReportInfo);
-  return { reportId: savedDetails.insertedId.toString(), dateUploaded: utilisationReportInfo.dateUploaded };
+  const updatedResult = await utilisationReportDetailsCollection.updateOne(filterForReportInReportPeriod, utilisationReportInfo);
```

Comment:
- I added `'bank.id': uploadedByUser.bank.id` to the filter
- I updated the controller to return a `404` if it could not find an existing report as that is now the failure case
- I updated `saveUtilisationReportDetails` to accept the report id as the first input argument (and updated the unit tests)

These changes will appear as part of merging in my other branch with e2e tests, so won't appear for a while. If it sounds like I've missed something, please let me know so I can sort that before merging that branch in